src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,121 @@
+// eslint-disable-next-line object-curly-newline
+import classNames from 'classnames';
+import { useEffect, useMemo, useRef } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useLocalStorage } from '../../hooks/useLocalStorage';
+import { Todo } from '../../types/Todo';
+import { Header } from '../Header';
+import { TodoList } from '../TodoList';
+import { TodosFilter } from '../TodosFilter';
+import { filterTotos } from '../../helpers/filterTotos';
+
+export const TodoApp: React.FC = () => {
+  const [todos, setTodos] = useLocalStorage('todos', []);
+
+  const completedTodosCount = useMemo(
+    () => todos.filter((todo: Todo) => todo.completed).length,
+    [todos],
+  );
+  const activeTodosCount = useMemo(
+    () => todos.filter((todo: Todo) => !todo.completed).length,
+    [todos],
+  );
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const isAllCompleted = completedTodosCount === todos.length;
+  const { pathname: location } = useLocation();
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, []);
+
+  const handleAddTodo = (text: string) => {
+    const newTodo: Todo = {
+      id: +new Date(),
+      title: text,
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
+  };
+
+  const toggleTodoAllHandler = () => {
+    setTodos(
+      todos.map((todo: Todo) => (
+        isAllCompleted
+          ? { ...todo, completed: false }
+          : { ...todo, completed: true })),
+    );
+  };
```

Comment:
separate this & use it in all cases
```suggestion
  const toggleTodoAllHandler = () => {
    setTodos(newTodos)
};
```